lib/sinon/util/core/wrap-method.js

```diff
@@ -106,49 +122,82 @@ module.exports = function wrapMethod(object, property, method) {
         simplePropertyAssignment();
     }
 
-    extend.nonEnum(method, {
-        displayName: property,
+    extendObjectWithWrappedMethods();
+
+    function extendObjectWithWrappedMethods() {
+        for (i = 0; i < wrappedMethods.length; i++) {
```

Comment:
Try to avoid using shared variables across functions (specifically `i` in this case), as that can lead to some difficult to spot bugs.

In this case, I think it is perfectly valid to use the ES5 Array Extras, i.e. `forEach`.


```suggestion
        wrappedMethods.forEach(function(wrapped){
```